### DISCUSSION TITLE
Inconsistent Metadata Sharing on Social Media (LinkedIn)

### DISCUSSION THREAD

**Discussion by jimburch** [gatsbyjs/gatsby#35123] [Help]
SOLVED: I switched my hosting from Heroku to Netlify and it fixed the issue

I use `react-helmet` to populate meta tags so pages on my site will render previews on social media. Most of it works normally but the one I need it working on most -- LinkedIn -- defaults to the home site metadata.

I pull in the site data via `graphql` inside an SEO component:

```
const Seo = (props) => {
  const data = useStaticQuery(graphql`
    query GetSiteMetadata {
      site {
        siteMetadata {
          title
          siteUrl
          image
          description
          author
        }
      }
    }
  `);

  const defaults = data.site.siteMetadata;

  const title = props.title || defaults.title;
  const description = props.description || defaults.description;
  const image = new URL(props.image || defaults.image, defaults.siteUrl);
  const url = new URL(props.path || "/", defaults.siteUrl);
  const author = props.author || defaults.author;

  return (
    <Helmet
      htmlAttributes={{
        lang: "en",
      }}
    >
      <title>{title}</title>
      <meta name="description" content={description} />
      <link rel="canonical" href={url} />
      <meta name="image" content={image} />
      <meta name="author" content={author} />

      <meta property="og:url" content={url} />
      <meta property="og:type" content="website" />
      <meta property="og:title" content={title} />
      <meta property="og:description" content={description} />
      <meta property="og:image" content={image} />

      <meta name="twitter:card" content="summary_large_image" />
      <meta name="twitter:title" content={title} />
      <meta name="twitter:description" content={description} />
      <meta name="twitter:image" content={image} />
    </Helmet>
  );
};
```

You can see where the component accepts props or defaults to the site metadata if there are none. I have a blog via MDX and pass down `frontmatter` data for each article:

```
const PostLayout = ({ children, pageContext }) => {
  const { title, description, date, metaImageUrl } = pageContext.frontmatter;

  const dateString = new Date(date).toLocaleDateString("en-US", {
    year: "numeric",
    month: "long",
    day: "numeric",
    timeZone: "UTC",
  });

  // ternary prevents error during build time
  const url = typeof window !== "undefined" ? window.location.href : "";

  return (
    <Layout
      title={`${title} | Changeblog`}
      description={description}
      image={metaImageUrl}
      path={url}
      style={{ paddingBottom: 50 }}
    >
      <h1>{title}</h1>
      <p>{dateString}</p>
      {children}
      <Link to="/changeblog">&larr; back to Changeblog</Link>
    </Layout>
  );
};

export default PostLayout;
```

The metadata populates just fine when I inspect the page, but here's where it gets interesting -- I have a staging environment on Heroku where the MDX path shows the right preview:

![image](https://user-images.githubusercontent.com/49823725/158101608-77abd93e-05b1-473a-9b2f-cc7939195cf8.png)
![image](https://user-images.githubusercontent.com/49823725/158101637-17a7adbf-3b22-45dc-838d-c8291ea92edd.png)

But the production URL defaults to the site metadata:

![image](https://user-images.githubusercontent.com/49823725/158101779-7ddc4a81-9920-434f-a528-b16318468a92.png)
![image](https://user-images.githubusercontent.com/49823725/158101799-896c0ca3-f815-47c7-8618-9f4bb2dc586e.png)

The only difference I can see between the two is that LinkedIn sees a canonical URL for the production environment. You can see in the code above I'm passing down the full `href` and it shows the full path in the inspector... even for the canonical. Weirdly enough, it's only LinkedIn that does this. Other share tools it works fine.

Staging: https://jimburch-portfolio-staging.herokuapp.com/changeblog/one-clean-landing-page-220228/
Prod: https://jimburch.com/changeblog/one-clean-landing-page-220228/

**Comment by miltonbolonha** [comment]
Hi @JimBurch.

Same problem for me, first thing until I fixed it I removed those two meta tags :og:url and canonical. It works temporarily.

**Reply by jimburch** [reply]
@miltonbolonha I'm not sure if Heroku was the root of the cause but I switched to Netlify and it solved this issue
